File: TypeScript/src/models/userModel.ts
import { Database } from "../database/db";

export class UserModel {
    static async create(username:string, password:string, name:string): Promise<void> {
        const db = await Database.getInstance();

        await db.execute(`CREATE TABLE IF NOT EXISTS users (
            username VARCHAR(255) NOT NULL,
            password VARCHAR(255) NOT NULL,
            name VARCHAR(255) NOT NULL,
            PRIMARY KEY (username)
            );`);
        await db.execute("INSERT INTO users (username, password, name) VALUES (?, ?, ?)", [username, password, name]);
    }

    static async findByUsernameAndPassword(username: string, password: string): Promise<any> {
        const db = await Database.getInstance();
        const [rows] = await db.execute("SELECT * FROM users WHERE username = ? AND password = ?", [username, password]);
        return rows[0];
    }
}

File: C#/Models/UserModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using TaskManagerApp.Database;

namespace TaskManagerApp.Models
{
    [Table("users")]
    public class Users
    {
        [Key]
        [Column("username")]
        [Required]
        public string? Username { get; set; }

        [Column("password")]
        [Required]
        public string? Password { get; set; }

        [Column("name")]
        public string? Name { get; set; }
    }


    public class UserModel
    {
        public static bool UserLogin(string username, string password, AppDbContext? context)
        {
            Users? result = context.Users.SingleOrDefault(e => e.Username == username && e.Password == password);

            if (result != null)
            {
                return true;
            }
            else
            {
                // Handle the case where no match is found (e.g., log an error, show a message to the user, etc.)
                return false;
            }
        }

        public static bool UserExistance(string username, AppDbContext? context)
        {
            Users? result = context.Users.SingleOrDefault(e => e.Username == username);

            if (result != null)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        public static bool Register(string username, string password, string name, AppDbContext context)
        {
            try
            {
                var user = new Users
                {
                    Username = username,
                    Password = password,
                    Name = name
                };

                context.Add(user);
                context.SaveChanges();
                return true;
            }
            catch (System.Exception)
            {
                return false;
            }
        }
    }
}